File: src/main/scala/ru/tinkoff/gatling/config/SimulationConfig.scala
package ru.tinkoff.gatling.config

import scala.concurrent.duration._
import ru.tinkoff.gatling.config.ConfigManager.simulationConfig
import ru.tinkoff.gatling.utils.IntensityConverter.getIntensityFromString

object SimulationConfig {

  def getStringParam(path: String): String           = simulationConfig.get[String](path)
  def getIntParam(path: String): Int                 = simulationConfig.get[Int](path)
  def getDoubleParam(path: String): Double           = simulationConfig.get[Double](path)
  def getDurationParam(path: String): FiniteDuration = simulationConfig.get[FiniteDuration](path)
  def getBooleanParam(path: String): Boolean         = simulationConfig.get[Boolean](path)

  lazy val baseUrl: String     = simulationConfig.get[String]("baseUrl")
  lazy val baseAuthUrl: String = simulationConfig.get[String]("baseAuthUrl")
  lazy val wsBaseUrl: String   = simulationConfig.get[String]("wsBaseUrl")

  lazy val stagesNumber: Int = simulationConfig.get[Int]("stagesNumber", 1)

  lazy val rampDuration: FiniteDuration  = simulationConfig.get[FiniteDuration]("rampDuration")
  lazy val stageDuration: FiniteDuration = simulationConfig.get[FiniteDuration]("stageDuration")
  lazy val testDuration: FiniteDuration =
    simulationConfig.get[FiniteDuration]("testDuration", (rampDuration + stageDuration) * stagesNumber)

  lazy val intensity: Double = getIntensityFromString(simulationConfig.get[String]("intensity"))

}

File: src/main/java/ru/tinkoff/gatling/javaapi/SimulationConfig.java
package ru.tinkoff.gatling.javaapi;

import java.time.Duration;

import static scala.compat.java8.DurationConverters.toJava;

public final class SimulationConfig {

    public static String getStringParam(String path) {
        return ru.tinkoff.gatling.config.SimulationConfig.getStringParam(path);
    }

    public static Integer getIntParam(String path) {
        return ru.tinkoff.gatling.config.SimulationConfig.getIntParam(path);
    }

    public static Double getDoubleParam(String path) {
        return ru.tinkoff.gatling.config.SimulationConfig.getDoubleParam(path);
    }

    public static Duration getDurationParam(String path) {
        return toJava(ru.tinkoff.gatling.config.SimulationConfig.getDurationParam(path));
    }

    public static Boolean getBooleanParam(String path) {
        return ru.tinkoff.gatling.config.SimulationConfig.getBooleanParam(path);
    }

    public static String baseUrl() {
        return ru.tinkoff.gatling.config.SimulationConfig.baseUrl();
    }

    public static String baseAuthUrl() {
        return ru.tinkoff.gatling.config.SimulationConfig.baseAuthUrl();
    }

    public static String wsBaseUrl() {
        return ru.tinkoff.gatling.config.SimulationConfig.wsBaseUrl();
    }

    public static Integer stagesNumber() {
        return ru.tinkoff.gatling.config.SimulationConfig.stagesNumber();
    }

    public static Duration rampDuration() {
        return toJava(ru.tinkoff.gatling.config.SimulationConfig.rampDuration());
    }

    public static Duration stageDuration() {
        return toJava(ru.tinkoff.gatling.config.SimulationConfig.stageDuration());
    }

    public static Duration testDuration() {
        return toJava(ru.tinkoff.gatling.config.SimulationConfig.testDuration());
    }

    public static Double intensity() {
        return ru.tinkoff.gatling.config.SimulationConfig.intensity();
    }
}
